weave

```diff
@@ -205,6 +205,40 @@ dns_arg_count() {
     fi
 }
 
+collect_rewrite_hosts_args() {
+    REWRITE_HOSTS_ARG_COUNT=0
+    case "$1" in
+    --rewrite-hosts)
+        REWRITE_HOSTS=1
+        REWRITE_HOSTS_ARG_COUNT=1
+        ;;
```

Comment:
it's even easier to inline there since you can just change the conditional that follows.